test/test_fingerprint_api.py

```diff
@@ -663,6 +679,75 @@ def test_get_related_visitors_429(self):
         self.assertEqual(context.exception.structured_error.error.code, ErrorCode.TOOMANYREQUESTS)
         self.assertEqual(context.exception.structured_error.retry_after, 4)
 
+    def test_search_events_only_limit(self):
+        """Test that search events returns 200 with only limit param"""
+        mock_pool = MockPoolManager(self)
+        self.api.api_client.rest_client.pool_manager = mock_pool
+        mock_pool.expect_request('GET', TestFingerprintApi.get_search_events_path(),
+                                 fields=[self.integration_info, ('limit', 1)],
+                                 headers=self.request_headers, preload_content=True, timeout=None)
+
+        response = self.api.search_events(1)
+        self.assertIsInstance(response, SearchEventsResponse)
+        event_response = response.events[0]
+        self.assertIsInstance(event_response, SearchEventsResponseEvents)
+        self.assertIsInstance(event_response.products, Products)
+        self.assertIsInstance(event_response.products.raw_device_attributes.data, RawDeviceAttributes)
+
+    def test_search_events_all_params(self):
```

Comment:
Does it make sense to also add test with partial params, to make sure that fields that were not provided are not sent to the API?

Reply:
Do you mean like `test_search_events_only_limit`, but with filling few optional too?

Reply:
Yep

Reply:
Done!